lxd/storage/s3/miniod/miniod.go

```diff
@@ -213,16 +217,70 @@ func EnsureRunning(bucketVol storageDrivers.Volume) (*Process, error) {
 		err := bucketVol.MountTask(func(mountPath string, op *operations.Operation) error {
 			l.Debug("MinIO bucket starting")
 
+			var newDirMode os.FileMode = os.ModeDir | 0700
+
 			if !shared.PathExists(bucketPath) {
-				err = os.Mkdir(bucketPath, 0700)
+				err = os.Mkdir(bucketPath, newDirMode)
+				if err != nil {
+					return fmt.Errorf("Failed creating MinIO bucket directory %q: %w", bucketPath, err)
+				}
+			}
+
+			dirInfo, err := os.Lstat(bucketPath)
+			if err != nil {
+				return fmt.Errorf("Failed getting MinIO bucket directory info %q: %w", bucketPath, err)
+			}
+
+			dirMode, dirUID, dirGID := shared.GetOwnerMode(dirInfo)
+
+			// Ensure file ownership is correct.
+			if uint32(dirUID) != s.OS.UnprivUID || uint32(dirGID) != s.OS.UnprivGID {
+				l.Debug("Setting MinIO bucket ownership", logger.Ctx{"currentOwner": dirUID, "currentGroup": dirGID, "newOwner": s.OS.UnprivUID, "newGroup": s.OS.UnprivGID})
+				err = filepath.Walk(bucketPath, func(path string, _ os.FileInfo, err error) error {
+					if err != nil {
+						return err
+					}
+
+					err = os.Chown(path, int(s.OS.UnprivUID), int(s.OS.UnprivGID))
+					if err != nil {
+						return fmt.Errorf("Failed setting ownership on MinIO bucket file %q: %w", path, err)
+					}
+
+					return nil
+				})
+				if err != nil {
+					return err
+				}
+			}
+
+			// Ensure permissions are correct.
+			if dirMode != newDirMode {
+				l.Debug("Setting MinIO bucket permissions", logger.Ctx{"currentMode": dirMode, "newMode": newDirMode})
+				err = os.Chmod(bucketPath, newDirMode)
 				if err != nil {
-					return fmt.Errorf("Failed creating MinIO bucket directory: %w", err)
+					return fmt.Errorf("Failed setting permissions on MinIO bucket directory %q: %w", bucketPath, err)
 				}
 			}
 
-			_, stdErr, err := shared.RunCommandSplit(minioProc.cancel, env, nil, "minio", args...)
+			cmd := exec.CommandContext(minioProc.cancel, "minio", args...)
+			cmd.Env = env
+
+			// Drop privileges.
+			cmd.SysProcAttr = &syscall.SysProcAttr{
+				Credential: &syscall.Credential{
```

Comment:
This should be `unix.SysProcAttr` and `unix.Credential`.

Reply:
Ah yes. Thanks

Reply:
I'll correct where I copied that from in the code base too

Reply:
Oh actually it cant be done because https://pkg.go.dev/golang.org/x/sys/unix#SysProcAttr is just an alias of https://pkg.go.dev/syscall#SysProcAttr and so setting the `Credential` field still requires importing `syscall`.